### DISCUSSION TITLE
Plugin Development

### DISCUSSION THREAD

**Discussion by example-user30** [GetPublii/Publii#1408] [Help] [answered]
Outside of dissecting an existing plugin - is there any documentation or help that has been written to help guide the process of creating a plugin?

I notice looking at the source code in most plugins that they are leveraging API hooks, such as this one from the imageDecoding Plugin:
`	this.API.addModifier`

Is there a list of available hooks and a general explanation of their use and parameters?

Thanks

**Comment by dziudek** [comment, marked as answer]
Hi,

I have started docs here: https://github.com/GetPublii/Publii/discussions/1359

After releasing v.0.45 in the beginning of the next week I will probably come back to expand this doc

**Reply by example-user30** [reply]
Wow!  Nice.  Much more than I knew trying to piece it together.  Thanks

**Reply by dziudek** [reply]
To be honest - it is only beginning 😎
